climada/hazard/tc_tracks_forecast.py

```diff
@@ -179,76 +184,156 @@ def fetch_bufr_ftp(target_dir=None, remote_dir=None):
 
         return localfiles
 
-    def read_one_bufr_tc(self, file, id_no=None, fcast_rep=None):
+    def read_one_bufr_tc(self, file, id_no=None):
         """ Read a single BUFR TC track file.
 
         Parameters:
             file (str, filelike): Path object, string, or file-like object
             id_no (int): Numerical ID; optional. Else use date + random int.
-            fcast_rep (int): Of the form 1xx000, indicating the delayed
-                replicator containing the forecast values; optional.
-        """
-
-        decoder = pybufrkit.decoder.Decoder()
-
-        if hasattr(file, 'read'):
-            bufr = decoder.process(file.read())
-        elif hasattr(file, 'read_bytes'):
-            bufr = decoder.process(file.read_bytes())
-        elif Path(file).is_file():
-            with Path(file).open('rb') as i:
-                bufr = decoder.process(i.read())
+        """     
+        # Open the bufr file
+        bufr = ec.codes_bufr_new_from_file(file)
+        # we need to instruct ecCodes to expand all the descriptors
+        # i.e. unpack the data values
+        ec.codes_set(bufr, 'unpack', 1)
+        
+        # get the forcast time
+        timestamp_origin = dt.datetime(
+            ec.codes_get(bufr, 'year'), ec.codes_get(bufr, 'month'),
+            ec.codes_get(bufr, 'day'), ec.codes_get(bufr, 'hour'),
+            ec.codes_get(bufr, 'minute'),
+        )
+        timestamp_origin = np.datetime64(timestamp_origin)
+        
+        # number of timestep
+        n_timestep = 0
+        while True:
+            n_timestep = n_timestep + 1
+            try:
+                ec.codes_get_array(bufr, "#%d#timePeriod" % n_timestep)
+            except ec.CodesInternalError:
+                LOGGER.info('Key values not found: end of the time period')
+                break
+            
+        # ensemble members number
+        ens_no = ec.codes_get_array(bufr, "ensembleMemberNumber")
+        
+        # initiate a data dictionary to hold values
+        latitude = collections.defaultdict(dict)
+        longitude = collections.defaultdict(dict)
+        pressure = collections.defaultdict(dict)
+        max_wind = collections.defaultdict(dict)
+        
+        # values at timestep 0
+        lat_init = ec.codes_get_array(bufr, '#2#latitude')
+        lon_init = ec.codes_get_array(bufr, '#2#longitude')
+        pre_init = ec.codes_get_array(bufr, '#1#pressureReducedToMeanSeaLevel')
+        max_wind_init = ec.codes_get_array(bufr, '#1#windSpeedAt10M')
+        
+        if len(lat_init) == len(ens_no) and len(max_wind_init) == len(ens_no):
+            for ind_ens in range(len(ens_no)):
+                latitude[ind_ens] = np.array(lat_init[ind_ens])
+                longitude[ind_ens] = np.array(lon_init[ind_ens])
+                pressure[ind_ens] = np.array(pre_init[ind_ens])
+                max_wind[ind_ens] = np.array(max_wind_init[ind_ens])
```

Comment:
Please use list (dict) comprehension here

Reply:
This has been updated.

Reply:
Replaced by list (dict) comprehension.